src/run.ts

```diff
@@ -1,39 +1,66 @@
 import * as vscode from 'vscode';
 import * as path from 'path';
+import * as findUp from 'find-up';
+import { configFiles } from './vitest-config-files';
 
-function buildVitestArgs(text: string) {
-    return ['vitest', 'run', '-t', text];
+function getCwd(testFile: string) {
+    const configFilePath = findUp.sync(configFiles, { cwd: testFile });
+
+    if (!configFilePath) {
+        return;
+    }
+    return path.dirname(configFilePath);
 }
 
-function buildCdArgs(path: string) {
-    return ['cd', path];
+function buildVitestArgs({ caseName, casePath, sanitize = true }: { caseName: string, casePath: string, sanitize?: boolean }) {
+    let sanitizedCasePath = casePath;
+    if (sanitize) {
+        sanitizedCasePath = JSON.stringify(casePath);
+        caseName = JSON.stringify(caseName);
+    }
+
+    const args = ['vitest', 'run', '-t', caseName, '--dir', sanitizedCasePath];
+
+    const rootDir = getCwd(casePath);
+    if (rootDir) {
+        args.push('--root', rootDir);
+    }
+
+    return args;
 }
 
+let terminal: vscode.Terminal | undefined;
+
 export function runInTerminal(text: string, filename: string) {
     const casePath = path.dirname(filename);
-    const terminal = vscode.window.createTerminal(`vitest - ${text}`);
+    let terminalAlreadyExists = true;
+    if (!terminal || terminal.exitStatus) {
+        terminalAlreadyExists = false;
+        terminal?.dispose();
+        terminal = vscode.window.createTerminal(`vscode-vitest-runner`);
+    }
```

Comment:
👍🏻 hope this change is released soon. Would like this behavior.

Reply:
forked and published my own package given this is not maintained https://marketplace.visualstudio.com/items?itemName=rluvaton.vscode-vitest